src/utils/vr-caps-detect.js

```diff
@@ -0,0 +1,81 @@
+const { detect } = require("detect-browser");
+const browser = detect();
+
+// Precision on device detection is fuzzy -- we can sometimes know if a device is definitely
+// available, or definitely *not* available, and assume it may be available otherwise.
+export const VR_DEVICE_AVAILABILITY = {
+  yes: "yes", // Implies VR can be launched into on this device immediately, in this browser
+  no: "no", // Implies this VR device is definitely not supported regardless of browser
+  maybe: "maybe" // Implies this device may support this VR platform, but may not be installed or in a compatible browser
+};
+
+function isMaybeGearVRCompatibleDevice() {
+  return navigator.userAgent.match(/\WAndroid\W/);
+}
+
+function isMaybeDaydreamCompatibleDevice() {
+  return navigator.userAgent.match(/\WAndroid\W/);
+}
+
+// Blacklist of VR device name regex matchers that we do not want to consider as valid VR devices
+// that can be entered into as a "generic" entry flow.
+const GENERIC_ENTRY_TYPE_DEVICE_BLACKLIST = [/cardboard/i];
```

Comment:
Why blacklist Cardboard? We may be cutting off a huge swath of people who currently enjoy WebVR content via Cardboard because they can't afford a legit VR device. I feel pretty strongly that we shouldn't limit access on those grounds.

Reply:
the thinking was less about cutting off cardboard users and more about ensuring that people don't accidentally go into cardboard mode if they have a gearvr or daydream. i'll think about how to avoid that without inadvertently cutting off cardboard users.

Reply:
Ah, okay. In theory you won't have to worry about the Daydream case. As long as we keep our origin trial up to date, we can detect browser exposing the WebVR APIs. With GearVR, we'll have to rely on UA parsing unfortunately

Reply:
i'm torn on this one. for people who are in a WebVR capable browser (Chrome or Samsung internet today) that own legit VR hardware, we'll be presenting them in many cases with a choice in the menu that is fundamentally going to head them in the wrong direction, confuse them, and possibly damage the app's perception.

Concrete example: I own a GearVR and have the page open in Chrome. I do not have Daydream installed. I will be presented with two choices: "Open in Cardboard" (since the device exists) and "Open on GearVR" (since we always show this because we can't detect it) -- the cardboard choice will have a VR headset icon next to it so many of these users will erroneously choose that option. They won't know that they made a wrong choice and the experience of it just going into lens distortion mode and not using their headset properly will look like, at best, a bug, at worst, normal and alienate them from thinking the app is solid. 

i'm leaning towards leaving things as-is -- one of my additional assumptions is that for ~100% of the people who engage cardboard mode their device is not going to hit every frame anyway so this will just make them sick, and they should be using magic window mode. do we know if that's a bad assumption?

Reply:
> ~100% of the people who engage cardboard mode their device is not going to hit every frame 

I think this is a bad assumption. There are many high-end phones out there that happen to not support Daydream or GearVR.

Furthermore, people who will opt to use Cardboard mode are likely to know what experience they're getting into. If they use a Cardboard regularly, they either have a good enough phone or they are very tolerant to VR sickness.

I would rather provide the Cardboard option at the end of the list when we can't be certain of Daydream/GearVR support. We can even design to indicate that it's the less desirable option if you prefer (e.g. no VR icon, greyed out button). 

I'd be fine with removing the Cardboard option if we prove otherwise after a couple of months of metrics, but I strongly disagree with cutting off that audience to begin with.

Reply:
Ok I think de-emphasizing it somehow might be a good way to do it. We already have an "Enter on another device" prompt in the design (for standalones, etc) so perhaps it can eventually be behind that screen. For now I can put the cardboard option in the place of that text until we build out the whole "duck.link" flow.